operator/src/main/java/oracle/kubernetes/operator/helpers/PodHelper.java

```diff
@@ -505,6 +538,120 @@ protected String getClusterName() {
       addStartupEnvVars(vars);
       return vars;
     }
+
+    abstract class ExporterContext {
+      int getWebLogicRestPort() {
+        return selectPortByProtocolName().orElse(selectPortFromList());
+      }
+
+      private Optional<Integer> selectPortByProtocolName() {
+        return getContainerPorts().stream()
+              .filter(p -> Objects.equals(scan.getAdminProtocolChannelName(), p.getName()))
+              .findFirst()
+              .map(V1ContainerPort::getContainerPort);
+      }
+
+      private int selectPortFromList() {
+        return Stream.of(getAdminPort(), getListenPort(), getSslListenPort())
+              .filter(Objects::nonNull)
+              .findFirst()
+              .orElseThrow(() -> new RuntimeException("No ports defined for this server"));
+      }
+
+      boolean isWebLogicSecure() {
+        return !Objects.equals(getWebLogicRestPort(), getListenPort());
+      }
+
+      abstract boolean isEnabled();
+
+      abstract int getPort();
+
+      abstract String getBasePath();
+
+      abstract void addContainer(List<V1Container> containers);
+    }
+
+    class WebAppExporterContext extends ExporterContext {
+
+      @Override
+      boolean isEnabled() {
+        return getListenPort() != null;
+      }
+
+      @Override
+      int getPort() {
+        return getListenPort();
+      }
+
+      @Override
+      String getBasePath() {
+        return "/wls-exporter";
+      }
+
+      @Override
+      void addContainer(List<V1Container> containers) {
+        // do nothing
+      }
+    }
+
+    class SidecarExporterContext extends ExporterContext {
+      private static final int DEBUG_PORT = 30055;
+      private final int metricsPort;
+
+      public SidecarExporterContext() {
+        metricsPort = MonitoringExporterSpecification.getRestPort(scan);
+      }
+
+      @Override
+      boolean isEnabled() {
+        return true;
+      }
+
+      @Override
+      int getPort() {
+        return metricsPort;
+      }
+
+      @Override
+      String getBasePath() {
+        return "";
+      }
+
+      @Override
+      void addContainer(List<V1Container> containers) {
+        containers.add(createMonitoringExporterContainer());
+      }
+
+      private V1Container createMonitoringExporterContainer() {
+        return new V1Container()
+              .name(EXPORTER_CONTAINER_NAME)
+              .image(getDomain().getMonitoringExporterImage())
+              .imagePullPolicy(getDomain().getMonitoringExporterImagePullPolicy())
+              .addEnvItem(new V1EnvVar().name("JAVA_OPTS").value(createJavaOptions()))
```

Comment:
I'm wondering if there's a need to populate server and domain-related environment variables that we populate in the main container and the init containers. I recently worked on populating these env variables in an init container in JIRA OWLS-77957.

Reply:
Shouldn't need them, as far as I know. The exporter only uses a small number of such variables.